var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -300,6 +300,19 @@ class QuantumEspresso(CMakePackage, Package):
 
     conflicts("@6.5:", when="+environ", msg="6.4.x is the latest QE series supported by Environ")
 
+    conflicts(
+        "@:7.3.0",
+        when="build_system=generic %oneapi",
+        msg="Support for ifx has been added to configure in release 7.3.1",
+    )
+    # TODO: When adding the next version make sure https://github.com/libmbd/libmbd/pull/60 is
+    # merged into the libmbd version used. Otherwise extend the conflict to the new version.
+    conflicts(
+        "@7.3.1 %oneapi@2024.1:",
+        when="build_system=cmake",
+        msg="ifx added intrinsic module ISO_C_BINDING in version 2024.1 which conflicts with the libmbd provided one.",
```

Comment:
"ifx added f_c_string in the ISO_C_BINDING module since version 2024.1 which conflicts with the libmbd provided one." to be accurate.

Reply:
Changed the message.